provider/infoblox/infoblox_test.go

```diff
@@ -393,50 +386,112 @@ func newInfobloxProvider(domainFilter endpoint.DomainFilter, zoneIDFilter provid
 	}
 }
 
+// TestInfobloxRecords function checks if the current records
+// are fetched properly by the ProviderConfig.Records method.
 func TestInfobloxRecords(t *testing.T) {
-	client := mockIBConnector{
-		mockInfobloxZones: &[]ibclient.ZoneAuth{
-			createMockInfobloxZone("example.com"),
-			createMockInfobloxZone("other.com"),
+	testCases := map[string]struct {
+		client            mockIBConnector
+		domainFilter      endpoint.DomainFilter
+		createPtr         bool
+		expectedEndpoints []*endpoint.Endpoint
+	}{
+		// A couple of CNAME/A records match the domain filter. Create PTR is disabled.
+		"CNAMEAndARecordsMatchDomainFilter": {
+			client: mockIBConnector{
+				mockInfobloxZones: &[]ibclient.ZoneAuth{
+					createMockInfobloxZone("example.com"),
+					createMockInfobloxZone("other.com"),
+				},
+				mockInfobloxObjects: &[]ibclient.IBObject{
+					createMockInfobloxObject("example.com", endpoint.RecordTypeA, "123.123.123.122"),
+					createMockInfobloxObject("example.com", endpoint.RecordTypeTXT, "heritage=external-dns,external-dns/owner=default"),
+					createMockInfobloxObject("nginx.example.com", endpoint.RecordTypeA, "123.123.123.123"),
+					createMockInfobloxObject("nginx.example.com", endpoint.RecordTypeTXT, "heritage=external-dns,external-dns/owner=default"),
```

Comment:
The TXT record format has been changed recently. We now may have 2 TXT records for each "target record": one as you already used (old format) and another one with the record type as prefix (new format).
I think the new format TXT records should be added to to make sure infoblox provider works with it too.